archive/control.go

```diff
@@ -854,3 +718,252 @@ func (control *Control) PollForDescriptors(correlationID int64, sessionID int64,
+				control.dispatchRecordingSignal()
+			}
+		}
+	}
+
+	// If we polled and did nothing then return
+	return aeron.NullValue, nil
+}
+
+// Poll for control response events. Returns number of fragments read during the operation.
+// Zero if no events are available.
+func (control *Control) Poll() (workCount int) {
+	if control.Results.IsPollComplete {
+		// Update our globals in case they've changed so we use the current state in our callback
+		rangeChecking = control.archive.Options.RangeChecking
+		control.Results = ControlResults{
+			ControlSessionId: aeron.NullValue,
+			CorrelationId:    aeron.NullValue,
+			RelevantId:       aeron.NullValue,
+			Version:          0,
+			Code:             codecs.ControlResponseCode.NullValue,
+			ErrorMessage:     []uint8{},
+
+			RecordingSignal: codecs.RecordingSignal.NullValue,
+			RecordingId:     aeron.NullValue,
+			SubscriptionId:  aeron.NullValue,
+			Position:        aeron.NullValue,
+
+			EncodedChallenge: []uint8{},
+
+			TemplateId:     aeron.NullValue,
+			IsPollComplete: false,
+
+			ControlResponse: nil,
+		}
+	}
+
+	return control.Subscription.ControlledPoll(control.fragmentAssembler.OnFragment, controlFragmentLimit)
+}
+
+// PollForResponse polls for a specific correlationID
+// Returns (relevantId, nil) on success, (-1 or relevantId, error) on failure
+// More complex responses are contained in Control.ControlResponse after the call
+func (control *Control) PollForResponse(correlationID int64, sessionID int64) (int64, error) {
+	logger.Debugf("PollForResponse(%d:%d)", correlationID, sessionID)
+
+	// Poll for events.
+	start := time.Now()
+
+	for {
+		err := control.pollNextResponse(start, correlationID, sessionID)
+		if err != nil {
+			logger.Debug(err) // log it in debug mode as an aid to diagnosis
+			return aeron.NullValue, err
+		}
+
+		if control.Results.ControlSessionId != control.archive.SessionID {
+			continue
+		}
+
+		// Check result
+		if control.Results.Code == codecs.ControlResponseCode.ERROR {
+			archiveErr := NewArchiveError(correlationID, int(control.Results.RelevantId), fmt.Sprintf("response for correlationId=%d, error: %s", correlationID, control.Results.ErrorMessage))
+			logger.Debug(archiveErr) // log it in debug mode as an aid to diagnosis
+
+			if control.Results.CorrelationId == correlationID {
+				return aeron.NullValue, archiveErr
+			}
+		}
+
+		if control.Results.CorrelationId == correlationID {
+			if control.Results.Code != codecs.ControlResponseCode.OK {
+				archiveErr := NewArchiveError(correlationID, int(control.Results.RelevantId), "unexpected response code: "+string(control.Results.Code))
+				logger.Debug(archiveErr) // log it in debug mode as an aid to diagnosis
+				return aeron.NullValue, archiveErr
+			}
+			return control.Results.RelevantId, nil
+		}
+	}
+}
+
+func (control *Control) pollNextResponse(startTime time.Time, correlationID, sessionID int64) error {
+	for {
```

Comment:
The idlestrategy reset. 

I think its as easy as `control.archive.Options.IdleStrategy.Idle(1)`